src/hooks/useInitialValues.test.ts

```diff
@@ -333,4 +378,63 @@ describe('useInitialValues', () => {
     expect(allFormFields.find((field) => field.id === 'testOrder_1')).not.toBeNull();
     expect(allFormFields.find((field) => field.id === 'testOrder_2')).not.toBeNull();
   });
+
+  it('should return calculated values for calculated fields in "edit" mode', async () => {
+    let hook = null;
+    const allFields = JSON.parse(JSON.stringify(calculateExpressionFields));
+    const allFieldsKeys = allFields.map((f) => f.id);
+    let valuesMap = {
+      height: '',
+      wight: '',
+      bmi: '',
+    };
+
+    const helper = new CommonExpressionHelpers(
+      { value: allFields[1], type: 'field' },
+      {},
+      allFields,
+      valuesMap,
+      allFieldsKeys,
+    );
+    await act(async () => {
+      hook = renderHook(() =>
+        useInitialValues(
+          [...allFormFields],
+          encounter,
+          false,
+          {
+            encounter: encounter,
+            patient: testPatient,
+            location,
+            sessionMode: 'enter',
+            encounterDate: encounterDate,
+            setEncounterDate: jest.fn,
+            encounterProvider: '2c95f6f5-788e-4e73-9079-5626911231fa',
+            setEncounterProvider: jest.fn,
+            setEncounterLocation: jest.fn,
+            encounterRole: '',
+            setEncounterRole: jest.fn,
+          },
+          formFieldHandlers,
+        ),
+      );
+    });
+    const {
+      current: { initialValues, isBindingComplete },
+    } = hook.result;
+    expect(isBindingComplete).toBe(true);
+
+    const weightQuestionConcept = calculateExpressionFields.find((field) => field.id === 'weight').questionOptions
+      .concept;
+    const heightQuestionConcept = calculateExpressionFields.find((field) => field.id === 'height').questionOptions
+      .concept;
+
+    const heightVal = encounter.obs.find((obs) => obs.concept.uuid === heightQuestionConcept).value;
+    const weightVal = encounter.obs.find((obs) => obs.concept.uuid === weightQuestionConcept).value;
+    const calculatedBmi = helper.calcBMI(heightVal, weightVal);
```

Comment:
Seems like you're manually evaluating the expressions and initial values here rendering the hook useless. You should be doing something like:

```ts
// verify BMI
expect(initialValues['bmi']).toBe(expectedValue);

// etc..
```